src/mongocxx/private/collection.hh

```diff
@@ -42,11 +42,13 @@ class collection::impl {
           client_impl{i.client_impl} {}
 
     impl& operator=(const impl& i) {
-        libmongoc::collection_destroy(collection_t);
-        collection_t = libmongoc::collection_copy(i.collection_t);
+        if (this != &i) {
```

Comment:
Coverity complained that this type wasn't protected against self assignment. It actually shouldn't be possible to ever self assign these, since they are always owned by unique_ptr, but just in case that rule ever changed, it seemed pretty harmless to add it. The code was definitely wrong if you did manage a self assignment.

Reply:
why not just be opinionated about this and delete the operator outright?

Reply:
Ah, because it does in fact get get invoked. Its just that I know that it will never get invoked as a self-assignment.